FlowCal/excel_ui.py

```diff
@@ -1054,18 +1083,44 @@ def add_samples_stats(samples_table, samples):
                                         channel + ' CV',
                                         FlowCal.stats.cv(sample, channel))
                 samples_table.set_value(row_id,
-                                        channel + ' Geom. Std',
-                                        FlowCal.stats.gstd(sample, channel))
-                samples_table.set_value(row_id,
-                                        channel + ' Geom. CV',
-                                        FlowCal.stats.gcv(sample, channel))
-                samples_table.set_value(row_id,
                                         channel + ' IQR',
                                         FlowCal.stats.iqr(sample, channel))
                 samples_table.set_value(row_id,
                                         channel + ' RCV',
                                         FlowCal.stats.rcv(sample, channel))
 
+                # For geometric statistics, first check for non-positive events.
+                # If found, throw a warning and calculate statistics on positive
+                # events only.
+                if np.any(sample[:, channel] <= 0):
+                    # Separate positive events
+                    sample_positive = sample[sample[:, channel] > 0]
+                    # Throw warning
+                    msg = "Geometric statistics for channel" + \
+                        " {} calculated on positive events".format(channel) + \
+                        " only ({:.1f}%). ".format(
+                            100.*sample_positive.shape[0]/sample.shape[0])
+                    warnings.warn("On sample {}: {}".format(row_id, msg))
+                    # Write warning message to table
+                    if samples_table.loc[row_id, 'Analysis Notes']:
+                        msg = samples_table.loc[row_id, 'Analysis Notes'] + msg
+                    samples_table.set_value(row_id, 'Analysis Notes', msg)
```

Comment:
we've talked previously about accumulating warnings throughout the duration of a particular analysis run. here, you seem to simply rely on concatenating warning strings (as opposed to a more explicit data structure for carrying around warnings)? is that correct? And you've padded all your warnings with a trailing space so they can concatenate in a general fashion?